File: CSharp/k07.cs
using System;

namespace CSharp
{
    class K07 : IExecutable
    {

        public override void Question01()
        {
            int choice = 0;
            do
            {
                try
                {
                    Console.Write("メッセージを表示しますか？（０：終了する　１：表示する）＞");
                    choice = int.Parse(Console.ReadLine());
                    if (choice == 1)
                    {
                        ShowTexts();
                    }
                }
                catch (Exception ex)
                {
                    Console.Error.WriteLine(ex.Message);
                }
                

            } while (choice != 0);
        }

        public override void Question02()
        {
            try
            {
                int[] numbers = new int[3];
                for (int i = 0; i < 3; i++)
                {
                    Console.Write(string.Format("{0}つ目の値を入力してください。＞", i + 1));
                    numbers[i] = int.Parse(Console.ReadLine());
                }

                Console.WriteLine(string.Format("{0}つの中で最大値は{1}です。",
                    numbers.Length, GetMaxValue(numbers[0], numbers[1], numbers[2])));
            }
            catch (Exception ex)
            {
                Console.Error.WriteLine(ex.Message);
            }
        }

        public override void Question03()
        {
            try
            {
                int age = 0;
                int no = 0;
                Console.Write("年齢を入力して下さい。＞");
                age = int.Parse(Console.ReadLine());

                no = this.GetType(age);
                switch (no)
                {
                    case 0:
                        Console.WriteLine("不適切な値が入力されました。");
                        break;
                    case 1:
                        Console.WriteLine("入場料金無料です。");
                        break;
                    case 2:
                        Console.WriteLine("子供料金で半額です。");
                        break;
                    case 3:
                        Console.WriteLine("シニア割引で１割引きです。");
                        break;
                    default:
                        Console.WriteLine("通常料金です。");
                        break;
                }
            }
            catch (Exception ex)
            {
                Console.Error.WriteLine(ex.Message);
            }
        }

        public override void Question04()
        {
            return;
        }

        private void ShowTexts()
        {
            Console.WriteLine("Hello World!");
            Console.WriteLine("ようこそ");
            Console.WriteLine("C#言語の世界へ！");
        }

        private int GetMaxValue(int value1, int value2, int value3)
        {
            int temp = value1 >= value2 ? value1 : value2;
            temp = temp >= value3 ? temp : value3;
            return temp;
        }

        private int GetType(int age)
        {
            if (age <= 0)
                return 0;
            else if (age < 3 || age >= 70)
                return 1;
            else if (age >= 3 && age <= 15)
                return 2;
            else if (age >= 60 && age < 70)
                return 3;
            else
                return 4;
        }
    }
}

File: Kotlin/src/K07.kt
import java.util.*

class K07 : IExecutable() {
    private fun show_text() {
        println("Hello World!")
        println("ようこそ")
        println("Kotlin言語の世界へ！")
    }

    private fun get_max_value(value1: Int, value2: Int, value3: Int): Int {
        var temp = if (value1 >= value2) value1 else value2
        temp = if (temp > value3) temp else value3
        return temp
    }

    private fun get_type(age: Int): Int {
        return when {
            (age <= 0) -> 0
            (age < 3 || age >= 70) -> 1
            (age in 3..15) -> 2
            (age in 60..69) -> 3
            else -> 4
        }
    }

    override fun question_01() {
        var choice: Int
        do {
            print("メッセージを表示しますか？（０：終了する　１：表示する）＞")
            val reader = Scanner(System.`in`)
            choice = reader.nextInt()
            if (choice == 1)
                show_text()
        } while (choice != 0)
        println("終了します。")
    }

    override fun question_02() {
        val numbers = Vector<Int>()
        val reader = Scanner(System.`in`)
        for (i in 1..3) {
            print("${i}つ目の値を入力してください。＞")
            numbers.add(reader.nextInt())
        }
        val max_value = get_max_value(numbers[0], numbers[1], numbers[2])
        println("${numbers.size}つの中で最大値は${max_value}です。")
    }

    override fun question_03() {
        val age: Int
        val no: Int
        print("年齢を入力して下さい。＞")
        val reader = Scanner(System.`in`)
        age = reader.nextInt()

        no = get_type(age)
        when (no) {
            0 -> println("不適切な値が入力されました。")
            1 -> println("入場料金無料です。")
            2 -> println("子供料金で半額です。")
            3 -> println("シニア割引で１割引きです。")
            else -> println("通常料金です。")
        }
    }

    override fun question_04() {
        return
    }
}